src/scripts/components/Timer.tsx

```diff
@@ -0,0 +1,156 @@
+import * as React from 'react';
+import styled from '@emotion/styled';
+import {
+  differenceInHours,
+  differenceInMinutes,
+  differenceInSeconds,
+  subHours,
+  subMinutes
+} from 'date-fns';
+
+import start from '../icons/start.svg';
+import stop from '../icons/stop.svg';
+
+const NO_DESCRIPTION = '(no description)';
+
+/**
+ * Returns the elapsed time since `since` as a duration in the format hh:mm:ss.
+ */
+export const formatDuration = (start: string | number, stop?: string | number) => {
+  const now = stop || Date.now();
+  const hours = differenceInHours(now, start);
+  const minutes = differenceInMinutes(subHours(now, hours), start);
+  const seconds = differenceInSeconds(subMinutes(subHours(now, hours), minutes), start);
+  const timeValue = (value) => value > 9 ? value : (value > 0 ? '0' + value : '00');
+
+  return `${timeValue(hours)}:${timeValue(minutes)}:${timeValue(seconds)}`;
+}
+
+type TimerProps = {
+  entry: TimeEntry | null;
+};
+
+function Timer (props: TimerProps) {
+  return props.entry
+    ? <RunningTimer entry={props.entry} />
+    : <TimerForm />
+}
+
+function RunningTimer(props: { entry: TimeEntry }) {
+  const { entry } = props;
+  const editEntry = (e) => {
+    e.preventDefault();
+    (window as any).PopUp.updateEditForm((window as any).PopUp.$editView);
+  };
+  const stopTimer = (e) => {
+    e.preventDefault();
+    (window as any).PopUp.sendMessage({ type: 'stop', service: 'dropdown', respond: true });
```

Comment:
Maybe instead of `window as any`, add to `index.d.ts`:
```js
interface Window {
  PopUp: any;
}
```
(for now. Then we improve its definition later)